src/main/java/duke/chatbot/ChatBot.java

```diff
@@ -0,0 +1,63 @@
+package duke.chatbot;
+
+import duke.data.TaskList;
+import duke.chatbot.command.Command;
+import duke.chatbot.command.ExitCommand;
+
+import java.util.ArrayList;
+
+/**
+ * ChatBot class that serves as an interface between
+ * the JavaFX GUI application and execution of user
+ * inputted commands.
+ */
+public class ChatBot {
+
+    private static final String WELCOME_STRING = "Hello I'm Duke!\n"
+            + "What can I do for you?";
+
+    /** Task list maintaining list of tasks for user */
+    private final TaskList taskList;
+
+    /** Boolean to track if bot has received a termination command */
+    private boolean hasTerminated;
+
+    public ChatBot(TaskList taskList) {
+        this.taskList = taskList;
+        this.hasTerminated = false;
+    }
+
+    /**
+     * Performs initialisation of the ChatBot. Should be called
+     * before receiving commands.
+     */
+    public void initialise() {
+        System.out.println(WELCOME_STRING);
+    }
+
+    /**
+     * Runs the command given by user input string, and returns
+     * a response from the ChatBot for the command, in the form
+     * of an ArrayList of string.
+     *
+     * @param input String command by user to run.
+     * @return ArrayList of string containing response of command ran.
```

Comment:
One suggestion, you could have followed one convention overall. Like you can begin all descriptions for `@params`, `@throws` and `@return` with capital letters.